packages/@react-aria/autocomplete/src/useAutocomplete.ts

```diff
@@ -152,11 +157,21 @@ export function useAutocomplete(props: AriaAutocompleteOptions, state: Autocompl
     }
 
     switch (e.key) {
+      case 'a':
+        if (isCtrlKeyPressed(e)) {
+          return;
+        }
+        break;
       case 'Escape':
         // Early return for Escape here so it doesn't leak the Escape event from the simulated collection event below and
         // close the dialog prematurely. Ideally that should be up to the discretion of the input element hence the check
         // for isPropagationStopped
+        // Also set the inputValue to '' to cover Firefox case where Esc doesn't actually clear searchfields. Normally we already
+        // handle this in useSearchField, but we are directly setting the inputValue on the input element in RAC Autocomplete instead of
+        // passing it to the SearchField via props. This means that a controlled value set on the Autocomplete isn't synced up with the
+        // SearchField until the user makes a change to the field's value via typing
         if (e.isPropagationStopped()) {
+          state.setInputValue('');
           return;
```

Comment:
As noted in the comment, we currently apply the tracked `inputValue` directly on to the Autocomplete wrapped input element rather than providing it via props to TextField/SearchField so that we can just consume the `inputValue` from the input context once rather than needing to add it to every input field component in RAC. This means that the input value tracked here in `useAutocomplete` isn't synced with the input value tracked in `useSearchField` until the user types into the field to cause the chained `onChange` to fire. Since Firefox doesn't also clear the native search input fields on `Esc` we won't get a `onChange` fired that clears the field so we'll need to do this here. Kinda gross, maybe reason enough to pass `value` to the input components via props (aka via TextFieldContext/SearchFieldContext) instead of applying it directly to the input element (aka via InputContext like we are doing right now)?